documentation/smg/16-serialiser.md

```diff
@@ -0,0 +1,275 @@
+# Serialisation details
+
+As of the integration of the C++ serialiser, the serialised format has been updated to be more consistent and to have greater capabilities.
+
+For reference, the old format is also detailed below
+
+## New format
+
+### Standard format
+The new style has a standardised header format depending on the shape of the data (with the exception of function handles).
+This format allows extension to support arrays of objects and structs in a way which the old serialiser couldn't.
+It also greatly simplifies the tag structure and allows it to align with the C++ MEX API data types.
+Data in these cases is serialised in the linearised order as defined by `data(:)` and the array reshaped by the deserialiser.
+
+This standard format for the header is as follows:
+RANK |     tag    | dim1 | dim2 | ... | Data
+-----|------------|------|------|-----|------
```

Comment:
I'm not finding this clear. What is RANK? Is each of these lines a possible header or do they all occur in one header. What determines N or the max number of headers or header rows? I'll look at the tag structure below (it seemed clear on a first glance). I'm presuming that the ellipsis ... stands for a possible number of dimensions, but you don't say. If so, an additional dimM column before data would make that a lot more clear.  What determines M? (Or, looking at the N row below, is N==M?) I was expecting Data to be separate from the header, following it as a body - does this mean that it is not really just a header but a whole serialized record?

Reply:
Rank is the array rank as in number of dimensions. The table details the whole header and data simply follows the header as a body, but the type and representation of the data is not described here.

Reply:
Would you rather I removed the "data" column here?

Reply:
As noted above, I think omitting the Data column would make it clearer that you are describing the header and not the data. I am also unclear why there is a separate rank column when that information is available in the tag column.

Reply:
The rank is not a part of the data header, but representative of the data, I should probably make that clearer.